### ISSUE TITLE
Shuffling causes line-pitch/pitch15 to fail

### ISSUE THREAD

**Issue by brunoabinader** [mapbox/mapbox-gl-native#9900]
Reproduced via:
```
$ node platform/node/test/render.test.js --recycle-map \
    icon-pitch-scaling/rotation-alignment-viewport \
    text-pitch-alignment/viewport-text-rotation-alignment-map \
    line-pitch/pitch15
* passed icon-pitch-scaling rotation-alignment-viewport
* ignore text-pitch-alignment viewport-text-rotation-alignment-map (https://github.com/mapbox/mapbox-gl-native/issues/9732)
* failed line-pitch pitch15
```

| expected | actual | diff |
|--|--|--|
| ![expected](https://user-images.githubusercontent.com/76133/29926839-384199b2-8e6d-11e7-8003-8e8304a1bdc2.png) | ![actual](https://user-images.githubusercontent.com/76133/29926826-3411ab98-8e6d-11e7-8e8c-d26bbcd1d037.png) | ![diff](https://user-images.githubusercontent.com/76133/29926852-3f791eee-8e6d-11e7-8acb-5f9f9786330b.png) |

_Possibly_ related: https://github.com/mapbox/mapbox-gl-native/issues/9732

**Comment by brunoabinader** [comment]
Hint: it has to do with the fact that `line-pitch/pitch15` has no sprite URL set. Adding a sprite URL to the style "fixes" it - but we shouldn't need to.

**Comment by brunoabinader** [comment]
The sprite URL affected the rendering test result because having a valid sprite URL would eventually load up the `building-12` image, causing relayout to happen because of this code: https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/renderer/renderer_impl.cpp#L208-L210

Proper fix in #9905.
